save-core/src/commonMain/kotlin/org/cqfn/save/core/Save.kt

```diff
@@ -72,11 +91,26 @@ class Save(
             }
 
         logInfo("Discovered plugins: $plugins")
+        plugins.forEach { plugin ->
+            reporter.onPluginInitialization(plugin)
+        }
+
         plugins.forEach { plugin ->
             logInfo("Execute plugin: ${plugin::class.simpleName}")
-            plugin.execute().forEach(this::handleResult)
+            reporter.onPluginExecutionStart(plugin)
+            try {
+                plugin.execute()
+                    .onEach { event -> reporter.onEvent(event) }
```

Comment:
it?

Reply:
I wanted to be more explicit, it could be shortened even to `.onEach(reporter::onEvent)`